### PR TITLE
fix: correct docker name for devenv images

### PR DESCRIPTION
**Pull request by yj-yan**
## What this PR does / why we need it

Fix the docker image name and tag for multiple containers in devenv.

## Jira ID

[DT-3923]

## Notes for your reviewers

Synced with @ygcao for the AWS permission, docker build and devenv deploy tested locally.

[DT-3923]: https://outreach-io.atlassian.net/browse/DT-3923?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shell/ci/release/docker.sh [hunk 1]
```diff
@@ -171,6 +165,26 @@ build_and_push_image() {
...
+  if [[ -n $DEVENV_DOCKER_BUILD ]]; then
+    info "Tagging the images for devenv"
+    if [[ "$image" != "$APPNAME" ]]; then
+      devenv_image="gcr.io/outreach-docker/$APPNAME/$image"
+      (
+        set -x
+        docker tag "$image":latest "$devenv_image":local
+      )
+    else
+      devenv_image="gcr.io/outreach-docker/$image"
```
Comment (jaredallard) [inline shell/ci/release/docker.sh:181]: `$remote_image_name` should suffice instead of hardcoding here, right?

File: shell/ci/release/docker.sh [hunk 2]
```diff
@@ -171,6 +165,26 @@ build_and_push_image() {
...
+  # Tag the images for devenv
+  # For primary container, the image should be built with gcr.io/outreach-docker/$APPNAME:latest
+  # For dependency container, the image should be built with gcr.io/outreach-docker/$APPNAME/$image:local 
+  if [[ -n $DEVENV_DOCKER_BUILD ]]; then
+    info "Tagging the images for devenv"
+    if [[ "$image" != "$APPNAME" ]]; then
+      devenv_image="gcr.io/outreach-docker/$APPNAME/$image"
+      (
+        set -x
+        docker tag "$image":latest "$devenv_image":local
```
Comment (jaredallard) [inline shell/ci/release/docker.sh:178]: Does this work? I believe that the `kind` loading logic in the devenv would also need to be changed?

File: shell/ci/release/docker.sh (hunk 1 above)
Comment (yj-yan) [reply shell/ci/release/docker.sh:181]: ahh, right. Good reminding.

File: shell/ci/release/docker.sh (hunk 2 above)
Comment (yj-yan) [reply shell/ci/release/docker.sh:178]: I run `devenv apps deploy .`, which seems to work. let me clear my kind cache and test again.

Comment (yj-yan) [reply shell/ci/release/docker.sh:178]: I ran it again on a newly provisioned devenv. The images were loaded on nodes. Got some unhealthy checks, but don't think it is related. hmm, now I wonder how the images were loaded
```
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  5m25s                  default-scheduler  Successfully assigned viamplify--bento1a/viamplify-d8cb96bc5-nmq9v to dev-environment-control-plane
  Normal   Pulled     5m24s                  kubelet            Container image "gcr.io/outreach-docker/viamplify:local" already present on machine
  Normal   Created    5m24s                  kubelet            Created container viamplify
  Normal   Created    5m23s                  kubelet            Created container duckling
  Normal   Created    5m23s                  kubelet            Created container co-service
  Normal   Started    5m23s                  kubelet            Started container co-service
  Normal   Pulled     5m23s                  kubelet            Container image "gcr.io/outreach-docker/duckling:20210318T214905-0bf91a411f2a1f1f808b98bc0dc579b61cdecaec" already present on machine
  Normal   Started    5m23s                  kubelet            Started container viamplify
  Normal   Pulled     5m23s                  kubelet            Container image "gcr.io/outreach-docker/viamplify/co-service:local" already present on machine
  Normal   Started    5m22s                  kubelet            Started container duckling
  Warning  Unhealthy  5m9s                   kubelet            Readiness probe failed: Get "http://10.244.0.158:9000/ping": dial tcp 10.244.0.158:9000: connect: connection refused
  ```

Comment (yj-yan) [reply shell/ci/release/docker.sh:178]: I guess devenv destroy doesn't cleanup images. I need to exec into control panel and manually remove the image.  I will address the kind loading part.

**Comment by yj-yan** [comment]
Drop this PR as the ci/release/docker.sh is not designed to build denenv images. I will make the build logic in devenv instead.

**Pull request closed without merging**